packages/eds-core-react/src/components/Datepicker/props.ts

```diff
@@ -0,0 +1,108 @@
+import { MutableRefObject, ReactNode } from 'react'
+import { CalendarState, RangeCalendarState } from '@react-stately/calendar'
+import { Variants } from '../types'
+import { HelperTextProps } from '../InputWrapper/HelperText'
+
+type DateRange = { from: Date | null; to: Date | null }
+
+export type HeaderProps<T = CalendarState | RangeCalendarState> = {
+  setYear: (year: number) => void
+  setMonth: (month: number) => void
+  year: number
+  month: number
+  state: T
+}
+
+export type DatePickerProps = Partial<{
+  /**
+   * The variant / state of the datepicker field
+   * @default undefined
+   */
+  variant: Variants
+  /**
+   * Whether the datepicker field is disabled
+   * @default false
+   */
+  disabled?: boolean
+  // eslint-disable-next-line react/no-unused-prop-types
+  readOnly?: boolean
+  /**
+   * Props to set the helper text
+   */
+  helperProps?: HelperTextProps
+  /**
+   * Controlled value
+   */
+  value: Date | null
+  /**
+   * The function to call when the datepicker field value changes
+   */
+  onChange: (date: Date | null) => void
+  /**
+   * The field label
+   */
+  label: string
+  /**
+   * minValue sets a limit to how far back the datepicker allows you to go
+   * note: this does not prevent you from typing in a date that is before the minValue
+   * @default undefined
+   */
+  minValue: Date
+  /**
+   * maxValue sets a limit to how far back the datepicker allows you to go
+   * note: this does not prevent you from typing in a date that is after the maxValue
+   * @default undefined
+   */
+  maxValue: Date
+  /**
+   * isDateUnavailable is used to disable specific dates (e.g. weekends)
+   * @param v
+   */
+  isDateUnavailable?: (v: Date) => boolean
+  /**
+   * An optional footer to display below the calendar, useful for e.g. presets
+   */
+  footer: ReactNode
+  /**
+   * An optional header to display above the calendar, can be used to override the default
+   * The component is fed with enough props to be able to control the calendar-view
+   * @param props
+   */
+  Header: (props: HeaderProps) => ReactNode
```

Comment:
Possibly; originally they were both just ReactNodes, but since the Header is used for navigating, I wanted to pass through some props. Could easily change footer to have the same signature.